pkg/ecdsa/tss/protocol_recovery.go

```diff
@@ -0,0 +1,147 @@
+package tss
+
+import (
+	"context"
+	cecdsa "crypto/ecdsa"
+	"fmt"
+	"time"
+
+	"github.com/keep-network/keep-core/pkg/net"
+)
+
+// BroadcastRecoveryAddress broadcasts and receives the BTC recovery addresses
+// of each client so that each client can retrieve the underlying bitcoin in
+// the case that a keep is terminated.
+func broadcastRecoveryAddress(
+	parentCtx context.Context,
+	groupID string,
+	memberID MemberID,
+	groupMemberIDs []MemberID,
+	dishonestThreshold uint,
+	networkProvider net.Provider,
+	pubKeyToAddressFn func(cecdsa.PublicKey) []byte,
+) error {
+	const protocolReadyTimeout = 2 * time.Minute
+
+	group := &groupInfo{
+		groupID:            groupID,
+		memberID:           memberID,
+		groupMemberIDs:     groupMemberIDs,
+		dishonestThreshold: int(dishonestThreshold),
+	}
+
+	netBridge, _ := newNetworkBridge(group, networkProvider)
+	broadcastChannel, _ := netBridge.getBroadcastChannel()
+	ctx, cancel := context.WithTimeout(parentCtx, protocolReadyTimeout)
+	defer cancel()
+
+	msgInChan := make(chan *LiquidationRecoveryMessage, len(group.groupMemberIDs))
+	handleLiquidationRecoveryMessage := func(netMsg net.Message) {
+		switch msg := netMsg.Payload().(type) {
+		case *LiquidationRecoveryMessage:
+			msgInChan <- msg
+		}
+	}
+	broadcastChannel.Recv(ctx, handleLiquidationRecoveryMessage)
+
+	memberBTCAddresses := make(map[string]string)
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case msg := <-msgInChan:
+				for _, memberID := range group.groupMemberIDs {
+					if msg.SenderID.Equal(memberID) {
+						memberAddress, err := memberIDToAddress(
+							memberID,
+							pubKeyToAddressFn,
+						)
+						if err != nil {
+							logger.Errorf(
+								"could not convert member ID to address for "+
+									"a member of keep [%s]: [%v]",
+								group.groupID,
+								err,
+							)
+							break
+						}
+						memberBTCAddresses[memberAddress] = msg.BtcRecoveryAddress
```

Comment:
As we move towards the multichain refactoring, it will probably make sense to make `memberAddress` be a `chain.OperatorID` and have the map key be that type as well.